### ISSUE TITLE
Firebase Auth with an Apple's identity token returns 17999 INVALID_CREDENTIAL_OR_PROVIDER_ID in a Xamarin.iOS app

### ISSUE THREAD

**Issue by alexsorokoletov** [xamarin/GoogleApisForiOSComponents#349]
I'm using [Xamarin.Firebase.iOS.Auth][1] and Apple [AuthenticationServices][2] to let a user to sign in using `Apple Sign In` and later authenticate with Firebase. I have a `Xamarin.iOS` app and the `Apple Sign In` part was done without any issues, I was able to activate `Apple Sign In` and receive a valid identity token from ASAuthorizationAppleIdCredential. The issue is in the second step, when I try to authenticate with Firebase using that token: 

    var idTokenString = "<my_valid_identity_token_here>";
    var credentials = OAuthProvider.GetCredential("apple.com", idTokenString);
    var authResult = await Auth.DefaultInstance.SignInWithCredentialAsync(credentials);

I'm getting an exception with the following details:

    (Inner Exception #0) Foundation.NSErrorException: Error Domain=FIRAuthErrorDomain Code=17999 "An internal error has occurred, print and inspect the error details for more information." UserInfo={FIRAuthErrorUserInfoNameKey=ERROR_INTERNAL_ERROR, NSLocalizedDescription=An internal error has occurred, print and inspect the error details for more information., NSUnderlyingError=0x600002c89260 {Error Domain=FIRAuthInternalErrorDomain Code=3 "(null)" UserInfo={FIRAuthErrorUserInfoDeserializedResponseKey={
        code = 400;
        errors =     (
                    {
                domain = global;
                message = "INVALID_CREDENTIAL_OR_PROVIDER_ID : Invalid IdP response/credential: http://localhost?providerId=apple.com&access_token=<my_token_here>";
                reason = invalid;
            }
        );

At first I thought that the token is invalid, but then I tried to use the same exact token received earlier (copy/pasted from xamarin app) in my native Swift example and I was able authenticate with Firebase without any issues:

    let appleIDToken =  "<my_valid_identity_token_here>";
    let credential = OAuthProvider.credential(
            withProviderID: "apple.com",
            idToken: appleIDToken,
            accessToken: nil)
            
    Auth.auth().signIn(with: credential, completion: { (authResult, error) in 
        // error = nil!, authResult - OK
    })

Both native and Xamarin apps are identical, same config, same bundle id, same capabilities. The only difference I found if the components version. Firebase native components are:

```
FirebaseCore (6.5.0)       vs    Xamarin.Firebase.iOS.Core (6.1.0)
FirebaseAuth (6.4.1)       vs    Xamarin.Firebase.iOS.Auth (6.2.1.1)  
FirebaseFirestore (1.8.3)  vs    Xamarin.Firebase.iOS.CloudFirestore (1.4.2.1)
```

Is it possible that the issue in the outdated versions provided by Xamarin or there could be another configuration issue specific to a Xamarin.iOS app?

  [1]: https://www.nuget.org/packages/Xamarin.Firebase.iOS.Auth/
  [2]: https://firebase.google.com/docs/auth/ios/apple

Also posted on SO: https://stackoverflow.com/questions/59434844/firebase-auth-attempt-to-use-an-apples-identity-token-returns-17999-invalid-cre

**Comment by alexsorokoletov** [comment]
@Redth thank you for taking look at the issue!
Just for planning purposes - when should we expect an update (range or approximation is ok)? 
I know it's Christmas and NY all around.

Thank you!

**Comment by alexsorokoletov** [comment]
@SotoiGhost maybe you can help or at least know who can help with this problem? it's been 23 days, I do not know whether I should wait or there are any other solutions/workarounds? Thank you

**Comment by alexsorokoletov** [comment]
I started updating the components myself and found the solution.
1. Xamarin component has different version numbers than Google Firebase SDK and Xamarin.Firebase.Core.iOS 6.1.0.0 version corresponds to FirebaseCore 6.5.0 which is the latest (https://github.com/xamarin/GoogleApisForiOSComponents/blob/master/components.cake#L122)

2. The issue in the application was related to the API interface definition and the following line passed the idTokenString value as access token

```
var credentials = OAuthProvider.GetCredential("apple.com", idTokenString);
```

So I changed to another overload and it worked well:

```
var credentials = OAuthProvider.GetCredential("apple.com", IdToken: idTokenString, accessToken: null);
```

This issue can be closed.
